code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/aepui/AepUI.kt

```diff
@@ -0,0 +1,51 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.aepcomposeui.aepui
+
+import com.adobe.marketing.mobile.aepcomposeui.aepui.state.AepUIState
+import com.adobe.marketing.mobile.aepuitemplates.AepUITemplate
+
+/**
+ * Represents a UI component that can be rendered using AEP UI templates.
+ * This is a sealed interface that can be implemented by different UI components like [SmallImageCardAepUi], [LargeImageCardAepUi], etc.
+ * It allows restricting the type of template and state that can be associated with a UI component, ensuring consistency
+ * when rendering and managing UI components across the app.
```

Comment:
Can we please remove the reasoning of why this is a sealed interface? We added this for our internal understanding only, no reason to have it in the public Kdocs